Many typo fixes

l-vo opened the pull request.

SenseException commented on docs/en/reference/second-level-cache.rst:
```diff
@@ -203,11 +195,11 @@ Cache Factory is the main point of extension.
+* ``QueryCache`` stores and retrieves query cache results.
+* ``CachedEntityPersister`` stores and retrieves entity results.
+* ``CachedCollectionPersister`` stores and retrieves query results.
+* ``EntityHydrator``  transforms an entity into a cache entry and caches entry into entities
```
```suggestion
* ``EntityHydrator``  transforms an entity into a cache entry and caches entries into entities
```
Because the sentence begins with "entity" (singular) and "entry" (singular), but ends with "entry" (singular) and "entities" (plural).

SenseException commented on docs/en/reference/second-level-cache.rst:
```diff
@@ -203,11 +195,11 @@ Cache Factory is the main point of extension.
-* ``CachedCollectionPersister`` Store and retrieve query results.
-* ``EntityHydrator``  Transform an entity into a cache entry and cache entry into entities
-* ``CollectionHydrator`` Transform a collection into a cache entry and cache entry into collection
+* ``QueryCache`` stores and retrieves query cache results.
+* ``CachedEntityPersister`` stores and retrieves entity results.
+* ``CachedCollectionPersister`` stores and retrieves query results.
+* ``EntityHydrator``  transforms an entity into a cache entry and caches entry into entities
+* ``CollectionHydrator`` transforms a collection into a caches entry and cache entry into collection
```
```suggestion
* ``CollectionHydrator`` transforms a collection into a cache entry and a cache entry into a collection
```
"a cache" as in "one cache".

greg0ire replied:
This could be reverted and converted to a description list (see http://html5doctor.com/the-dl-element/)

```suggestion
``QueryCache``
    stores and retrieves query cache results.
``CachedEntityPersister``
    stores and retrieves entity results.
``CachedCollectionPersister``
    stores and retrieves query results.
``EntityHydrator``
    transforms an entity into a cache entry and caches entry into entities
``CollectionHydrator``
    transforms a collection into a caches entry and cache entry into collection
```

SenseException commented on docs/en/reference/second-level-cache.rst:
```diff
@@ -225,8 +217,8 @@ To specify a default lifetime for all regions or specify a different lifetime fo
-    $regionConfig->setLifetime('my_entity_region', 3600);   // Time to live for a specific region; In seconds
-    $regionConfig->setDefaultLifetime(7200);                // Default time to live; In seconds
+    $regionConfig->setLifetime('my_entity_region', 3600);   // Time to live for a specific region (in seconds)
+    $regionConfig->setDefaultLifetime(7200);                // Default time to live  (in seconds)
```
```suggestion
    $regionConfig->setDefaultLifetime(7200);                // Default time to live (in seconds)
```

SenseException commented:
Can you please also rephrase your commit message? Be more descrptive what you fixed in a commit body.

greg0ire commented on docs/en/reference/second-level-cache.rst:
```diff
@@ -79,23 +79,17 @@ Cache Regions
-
-``Doctrine\ORM\Cache\Region`` and ``Doctrine\ORM\Cache\ConcurrentRegion``
-Defines contracts that should be implemented by a cache provider.
+``Doctrine\ORM\Cache\Region`` and ``Doctrine\ORM\Cache\ConcurrentRegion`` define contracts that should be implemented by a cache provider.
```
I think you should change line 81 too if we are going that way, also, there should be linebreaks here.

Please comply with the sf docs standards (I don't know if we follow them, but this particular rule seems appropriate):

> [Each line should break approximately after the first word that crosses the 72nd character (so most lines end up being 72-78 characters);](http://symfony.com/doc/current/contributing/documentation/standards.html)

greg0ire commented on docs/en/reference/second-level-cache.rst:
```diff
@@ -79,23 +79,17 @@ Cache Regions
 It allows you to provide your own cache implementation that might take advantage of specific cache driver.
 
-If you want to support locking for ``READ_WRITE`` strategies you should implement ``ConcurrentRegion``; ``CacheRegion`` otherwise.
+If you want to support locking for ``READ_WRITE`` strategies you should implement ``ConcurrentRegion`` (``CacheRegion`` otherwise).
```
Why change this? the second part is not a precision, it's the counterpart of the first one, and has a similar importance.

l-vo replied:
I reverted this change

greg0ire commented on docs/en/reference/second-level-cache.rst:
```diff
@@ -267,8 +259,7 @@ By providing a cache logger you should be able to get information about all cach
 
-If you want to get more information you should implement ``\Doctrine\ORM\Cache\Logging\CacheLogger``.
-and collect all information you want.
+If you want to get more information you should implement ``\Doctrine\ORM\Cache\Logging\CacheLogger`` and collect all information you want.
```
Please use linebreaks

greg0ire commented on docs/en/reference/second-level-cache.rst:
```diff
@@ -277,8 +268,8 @@ Entity cache definition
 
-  * ``usage`` Specifies the caching strategy: ``READ_ONLY``, ``NONSTRICT_READ_WRITE``, ``READ_WRITE``. see :ref:`reference-second-level-cache-mode`
-  * ``region`` Optional value that specifies the name of the second level cache region.
+  * ``usage`` specifies the caching strategy: ``READ_ONLY``, ``NONSTRICT_READ_WRITE``, ``READ_WRITE``. See :ref:`reference-second-level-cache-mode`.
```
linebreaks

l-vo committed:
Fix many typos

- Some uppercase letters were used in the middle of sentence
- Some dots were missing
- There was two sentences with wrong or missing words

l-vo commented:
@greg0ire @SenseException all threads are resolved.

greg0ire commented on docs/en/reference/second-level-cache.rst:
```diff
@@ -728,4 +732,5 @@ Paginator
 
 Count queries generated by ``Doctrine\ORM\Tools\Pagination\Paginator`` are not cached by second-level cache.
-Although entities and query result are cached count queries will hit the database every time.
+Although entities and query result are cached,count queries will hit the
```
```suggestion
Although entities and query result are cached, count queries will hit the
```

greg0ire commented on docs/en/reference/second-level-cache.rst:
```diff
@@ -622,7 +626,7 @@ Using the repository query cache
 
 As well as ``Query Cache`` all persister queries store only identifier values for an individual query.
-All persister use a single timestamps cache region keeps track of the last update for each persister,
+All persister use a single timestamp cache region to keep track of the last update for each persister,
```
```suggestion
All persisters use a single timestamp cache region to keep track of the last update for each persister,
```

greg0ire commented on docs/en/reference/second-level-cache.rst:
```diff
@@ -267,8 +266,9 @@ By providing a cache logger you should be able to get information about all cach
-and collect all information you want.
+If you want to get more information you should implement
+``\Doctrine\ORM\Cache\Logging\CacheLogger`` and collect
+all information you want.
```
```suggestion
all the information you want.
```

greg0ire commented on docs/en/reference/second-level-cache.rst:
```diff
@@ -203,11 +197,16 @@ Cache Factory is the main point of extension.
+``EntityHydrator``
+    transforms an entity into a cache entry and caches entry into entities
+``CollectionHydrator``
+    transforms a collection into a caches entry and cache entry into collection
```
```suggestion
    transforms collections into cache entries and cache entries into collections
```

greg0ire commented on docs/en/reference/second-level-cache.rst:
```diff
@@ -203,11 +197,16 @@ Cache Factory is the main point of extension.
+``CachedCollectionPersister``
+    stores and retrieves query results.
+``EntityHydrator``
+    transforms an entity into a cache entry and caches entry into entities
```
```suggestion
    transforms entities into a cache entries and cache entries into entities
```

greg0ire commented on docs/en/reference/second-level-cache.rst:
```diff
@@ -107,9 +103,7 @@ By default, Doctrine provides a very simple implementation based on file locks `
-``Doctrine\ORM\Cache\ConcurrentRegion``
-
-Defines contract for concurrently managed data region.
+``Doctrine\ORM\Cache\ConcurrentRegion`` defines contract for concurrently managed data region.
```
```suggestion
``Doctrine\ORM\Cache\ConcurrentRegion`` defines a contract for concurrently managed data region.
```

l-vo commented:
All threads are resolved
